### PR TITLE
FINERACT-1816: Create client with datatable subentity type

### PR DESCRIPTION
**Pull request by josehernandezfintecheandomx**
## Description

When a datatable is linked to Client creation it will be visible regardless of its subtype and the set Legal type
So for eg. we have two datatables, one for PERSON and one for ENTITY, both will be visible upon client creation wether the Legal form is set to PERSON or ENTITY

[FINERACT-1816](https://issues.apache.org/jira/browse/FINERACT-1816)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by galovics** [comment]
@josehernandezfintecheandomx can you pls change the commit message & PR title to have a Fineract ticket in it? Thanks.

**Review by galovics — CHANGES_REQUESTED** [review]
Looks good but I don't see tests for the changed functionality. Please add them.

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java [hunk 1]
```diff
@@ -263,4 +282,51 @@ public void testClientAddressCreationWorksAfterClientIsCreated() {
...
+        String datatabelRequestJsonString = new Gson().toJson(columnMap);
+        LOG.info("map : {}", datatabelRequestJsonString);
+
+        HashMap<String, Object> datatableResponse = datatableHelper.createDatatable(datatabelRequestJsonString, "");
+        final String resourceIdentifier = (String) datatableResponse.get("resourceIdentifier");
+        DatatableHelper.verifyDatatableCreatedOnServer(requestSpec, responseSpec, resourceIdentifier);
+
+        final PostEntityDatatableChecksTemplateResponse entityPersonDatatableChecksResponse = entityDatatableChecksHelper
+                .addEntityDatatableCheck(CLIENT_APP_TABLE_NAME, datatableName, 100, null);
+        assertNotNull(entityPersonDatatableChecksResponse);
```
Comment (galovics) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: Let's check more things from the response body, not just that it's not null.

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java [hunk 2]
```diff
@@ -263,4 +282,51 @@ public void testClientAddressCreationWorksAfterClientIsCreated() {
...
+        final HashMap<String, Object> datatablesMap = new HashMap<>();
+        datatablesMap.put("registeredTableName", datatableName);
+        datatablesMap.put("data", datatableEntryMap);
+
+        String datatablesJsonString = new Gson().toJson(datatablesMap);
+        LOG.info("map1 : {}", datatablesJsonString);
+
+        PostClientsResponse postClientsResponse = ClientHelper.createClientAsPersonWithDatatable(requestSpec, responseSpec, "04 March 2011",
+                "1", datatablesMap);
+        assertNotNull(postClientsResponse);
```
Comment (galovics) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: Let's check more things from the response body, not just that it's not null.

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java [hunk 3]
```diff
@@ -263,4 +282,51 @@ public void testClientAddressCreationWorksAfterClientIsCreated() {
...
+
+        postClientsResponse = ClientHelper.createClientAsPersonWithDatatable(requestSpec, responseSpecErr403, "04 March 2011", "1", null);
```
Comment (galovics) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: Let's check the response here.

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java [hunk 4]
```diff
@@ -263,4 +282,51 @@ public void testClientAddressCreationWorksAfterClientIsCreated() {
...
+
+        entityDatatableChecksHelper.deleteEntityDatatableCheck(entityPersonDatatableChecksResponse.getResourceId().intValue());
```
Comment (galovics) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: I think we're missing an assertion after this; that the entity datatable check has been really deleted.

**Review by galovics — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java (hunk 3 above)
Comment (josehernandezfintecheandomx) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: There is not any response, there are an exception for the missing (null) data for datatables

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java (hunk 1 above)
Comment (josehernandezfintecheandomx) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: Why do I need to validate this? If the change is related to Client creation with datatable data

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java (hunk 2 above)
Comment (josehernandezfintecheandomx) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: Why do I need to validate this? If the change is related to Client creation with datatable data

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java (hunk 4 above)
Comment (josehernandezfintecheandomx) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientTest.java]: Why do I need to validate this? If the change is related to Client creation with datatable data

**Review by galovics — APPROVED** [review]

**Pull request merged**